bench_test.go

```diff
@@ -0,0 +1,109 @@
+package frostdb
+
+import (
+	"context"
+	"testing"
+
+	"github.com/apache/arrow/go/v8/arrow"
+	"github.com/apache/arrow/go/v8/arrow/memory"
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+
+	"github.com/polarsignals/frostdb/query"
+	"github.com/polarsignals/frostdb/query/logicalplan"
+)
+
+const (
+	dbName      = "parca"
+	tableName   = "stacktraces"
+	storagePath = "testdata/data"
+)
+
+// This file runs end-to-end benchmarks on a single FrostDB instance using
+// Polar Signals-specific data and schema. These benchmarks are skipped by
+// default since they require a manually-generated data directory to run. To run
+// these benchmarks, unskip the benchmarks and create a data directory in
+// testdata/data.
+
+func newDBForBenchmarks(ctx context.Context, b *testing.B) (*DB, error) {
+	b.Helper()
+
+	b.Logf("initializing %s", b.Name())
+
+	col, err := New(
+		log.NewNopLogger(),
+		prometheus.NewRegistry(),
+		WithWAL(),
+		WithStoragePath(storagePath),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := col.ReplayWALs(ctx); err != nil {
+		return nil, err
+	}
+
+	colDB, err := col.DB(ctx, dbName)
+	if err != nil {
+		return nil, err
+	}
+
+	table, err := colDB.GetTable(tableName)
+	if err != nil {
+		return nil, err
+	}
+	table.Sync()
+
+	b.Logf("db initialized and WAL replayed, starting benchmark %s", b.Name())
+	return colDB, nil
+}
+
+// Remove unused warning.
+var _ = newDBForBenchmarks
+
+func BenchmarkTypes(b *testing.B) {
+	b.Skip()
```

Comment:
`nit` add the skip comment of "require data directory" or something along those lines